Demo/ChartIQApp/Models/Models/IntervalModel.swift

```diff
@@ -125,15 +140,18 @@ struct IntervalModel: Equatable, Codable {
 
   internal func getPeriod() -> Int {
     if timeUnit == .hour {
-      return time * 2
+      return calculatePeriod()
     } else {
-      return 1
+      return calculateInterval()
```

Comment:
@stoliarenkoks, so I think we need add some sort of isIntraday check so that the previous functionality will still apply to timeunits for millisecond, second, and minute. The check could be as simple as putting the other intraday values along with .hour.

The chart performance will degrade if a large custom value is entered for intraday values. Try entering 99 minutes into the custom periodicity and you'll see what I mean. Non-intraday, such as days, doesn't have that issue as there usually isn't that much data to roll up 99 day bars.

Reply:
@jdrichards50 At the moment, I did as you wrote in the letter, for the hour everything remains as it was, and for the rest, it changes places.
Let's discuss what else we need to do, together with Dmitry (Android), since he will have to do the same so that we are all be on the same page

Reply:
Okay, I'll set up a meeting.